internal/commands/sast-prompt.go

```diff
@@ -135,34 +138,22 @@ func GetMethodByMethodLine(filename string, lines []string, methodLineNumber, no
 	return methodLines, nil
 }
 
-func AddNewlinesIfNecessary(responseContent []string) []string {
-	if len(responseContent) == 0 {
-		return responseContent
-	}
-	stringToFix := responseContent[len(responseContent)-1]
-
-	stringToFix = addNewlineIfNecessary(stringToFix, confidence, explanation)
-	stringToFix = addNewlineIfNecessary(stringToFix, explanation, fix)
-	return append(responseContent[:len(responseContent)-1], stringToFix)
-}
-
-func addNewlineIfNecessary(s, from, to string) string {
-	startsAt := strings.Index(s, from) + len(from)
-	upTo := strings.Index(s, to)
-	if startsAt == -1 || upTo == -1 {
-		return s
+func addDescriptionForIdentifier(responseContent []string) []string {
+	identifiersDescription := map[string]string{
+		confidence:  " A score between 0 (low) and 100 (high) indicating OpenAI's confidence level for the effectiveness of the suggested remediation. <br>",
+		explanation: " An OpenAI generated description of the vulnerability. <br>",
+		fix:         " A customized snippet, generated by OpenAI, that can be used to remediate the vulnerability in your code. <br>",
 	}
-	if !endsWithNewlineAndWhitespace(s[startsAt:upTo]) {
-		return s[:upTo] + "\n" + s[upTo:]
+	identifierTitleFormat := "<span style=\"color: regular;\">%s</span><span style=\"color: grey; font-style: italic;\">%s</span>"
+	if len(responseContent) > 0 {
+		for i := 0; i < len(responseContent); i++ {
+			line := responseContent[i]
+			for identifier, description := range identifiersDescription {
+				line = strings.Replace(line, identifier, fmt.Sprintf(identifierTitleFormat, identifier, description), 1)
```

Comment:
is there a chance of a null reference exception here?

Reply:
No, in case it does not find the identifier it returns the original string